rcp/org.testeditor.rcp4.views.tests/src/org/testeditor/rcp4/views/tests/tree/AmlModelLabelProviderTest.xtend

```diff
@@ -0,0 +1,91 @@
+package org.testeditor.rcp4.views.tests.tree
+
+import javax.inject.Inject
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure1
+import org.junit.Test
+import org.testeditor.aml.AmlFactory
+import org.testeditor.aml.TemplateText
+import org.testeditor.aml.TemplateVariable
+import org.testeditor.rcp4.views.AmlModelLabelProvider
+
+class AmlModelLabelProviderTest extends AbstractParserTest {
+
+	@Inject
+	AmlModelLabelProvider amlModelLabelProvider
+
+// TODO: get this injected!
+//	@Inject
+//	AmlFactory factory
+	AmlFactory factory = AmlFactory.eINSTANCE
+
+	@Test
+	def void labelForModelIsPackage() {
+		// given
+		val amlModel = factory.createAmlModel
+		amlModel.^package = "test"
+
+		// when
+		val result = amlModelLabelProvider.getText(amlModel)
+
+		// then
+		assertEquals(result, "test")
+	}
+
+	@Test
+	def void labelForComponentIsName() {
+		// given
+		val component = factory.createComponent
+		component.name = "test"
+
+		// when
+		val result = amlModelLabelProvider.getText(component)
+
+		// then
+		assertEquals(result, "test")
+	}
+
+	@Test
+	def void labelForComponentElementIsName() {
+		// given
+		val componentElement = factory.createComponentElement
+		componentElement.name = "test"
+
+		// when
+		val result = amlModelLabelProvider.getText(componentElement)
+
+		// then
+		assertEquals(result, "test")
+	}
+
+	@Test
+	def void labelForInteractionTypesIsTheTemplateString() {
+		// given
+		val interactionType = factory.createInteractionType
+		interactionType.template = factory.createTemplate
+		interactionType.template.contents += #[
+			newTemplateText[value = "Insert Text"],
+			newTemplateVariable[name = "text"],
+			newTemplateText[value = "into"],
+			newTemplateVariable[name = "element"]
+		]
+
+		// when
+		val result = amlModelLabelProvider.getText(interactionType)
+
+		// then
+		assertEquals(result, "Insert Text \"text\" into <element>")
+	}
+
+	private def TemplateText newTemplateText(Procedure1<TemplateText> init) {
```

Comment:
Both `TemplateText` and `TemplateVariable` only have a single variable. We should pass the value as a regular parameter instead of passing a procedure for initialization.